src/main/java/com/softserve/sportshub/comment/service/CommentServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.softserve.sportshub.comment.service;
+
+
+import com.softserve.sportshub.comment.dao.CommentDao;
+import com.softserve.sportshub.comment.domain.Comment;
+import com.softserve.sportshub.comment.domain.CommentMapper;
+import com.softserve.sportshub.comment.dto.CommentDto;
+import com.softserve.sportshub.comment.dto.CreateCommentDto;
+import com.softserve.sportshub.comment.dto.ReadCommentDto;
+import com.softserve.sportshub.user.User;
+import com.softserve.sportshub.user.UserDao;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+@Service
+public class CommentServiceImpl implements CommentService {
+
+
+    private final CommentDao commentDao;
+    private final UserDao userDao;
+    private final CommentMapper mapper;
+
+    public CommentServiceImpl(CommentDao commentDao, UserDao userDao, CommentMapper mapper) {
+        this.commentDao = commentDao;
+        this.userDao = userDao;
+
+        this.mapper = mapper;
+    }
+
+    @Transactional
+    public ReadCommentDto save(Comment comment) {
+        commentDao.save(comment);
+        return mapper.entityToReadDto(comment);
+    }
+
+    @Override
+    @Transactional
+    public ReadCommentDto save(CreateCommentDto createCommentDto, String username) {
+        Comment comment = new Comment();
+        comment.setContent(createCommentDto.getContent());
+        comment.setCreatedAt(LocalDate.now().toString());
+        comment.setOwner(userDao.findByUsername(username));
+        comment.setListOfUserLikes(new HashSet<>());
+        comment.setListOfUserDislikes(new HashSet<>());
+        commentDao.save(comment);
+        return mapper.entityToReadDto(comment);
+    }
+
+
+    @Transactional(readOnly = true)
+    public List<ReadCommentDto> list() {
+        return commentDao.list().stream().map(mapper::entityToReadDto).toList();
+    }
+
+    @Override
+    @Transactional
+    public Comment findById(Long id) {
+        return commentDao.findById(id);
+    }
+
+    @Override
+    @Transactional
+    public boolean delete(Long id) {
+        return commentDao.delete(id);
+    }
+
+    @Transactional
+    public ReadCommentDto like(Long id, String username) {
+        User userLiked = userDao.findByUsername(username);
+        Comment likedComment = commentDao.findById(id);
+        if (likedComment.getOwner().getUsername().equals(username)) {
+            return mapper.entityToReadDto(likedComment);
+        }
+        likedComment.getListOfUserDislikes().remove(userLiked);
+        likedComment.getListOfUserLikes().add(userLiked);
+        commentDao.save(likedComment);
+        return mapper.entityToReadDto(likedComment);
+    }
```

Comment:
I'm not sure what the business requirements are (this may be valid) but currently one user can like a comment multiple times. Is it expected to work that way?

Reply:
User likes and dislikes are held in Set<User>, so user can like the comment multiple times but the likes counter will count only one like from this user.